src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,108 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        int nodeHash = key != null ? key.hashCode() : 0;
+        int nodeIndex = Math.abs(nodeHash) % table.length;
+        Node<K, V> node = table[nodeIndex];
+        if (node == null) {
+            table[nodeIndex] = new Node<>(nodeHash, key, value, null);
+            size++;
+            return;
+        }
+        while (true) {
+            if (node.key == key
+                    || node.key.equals(key)) {
+                node.value = value;
+                return;
+            }
+            if (node.next == null) {
+                node.next = new Node<>(nodeHash, key, value, null);
+                size++;
+                return;
+            }
+            node = node.next;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int nodeHash = key != null ? key.hashCode() : 0;
+        int nodeIndex = Math.abs(nodeHash) % table.length;
+        Node<K, V> node = table[nodeIndex];
+        while (node != null) {
+            if (node.key == key
+                    || node.key.equals(key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        size = 0;
+        Node<K, V>[] oldTable = table;
+        table = new Node[table.length * 2];
+        threshold = (int) (table.length * LOAD_FACTOR);
+        for (Node<K, V> node : oldTable) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
+
+    private static class Node<K, V> {
+        private final int hash;
```

Comment:
The idea is great to store key in your Node. But it is not used in your code

Reply:
removed